homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,334 @@
+                                      origin,
+                                      destination,
+                                      here_data)
+
+        hass.data[DATA_KEY].append(sensor)
+
+        add_entities_callback([sensor])
+
+    # Wait until start event is sent to load this component.
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(self, hass, name, origin, destination, here_data):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._name = name
+        self._here_data = here_data
+        self._unit_of_measurement = 'min'
+
+        # Check if location is a trackable entity
+        if origin.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._here_data.origin = origin
+
+        if destination.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._destination_entity_id = destination
+        else:
+            self._here_data.destination = destination
+
+        self.update()
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._here_data.duration is not None:
+            return round(self._here_data.duration / 60)
+
+        return None
+
+    @property
+    def name(self):
+        """Get the name of the sensor."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._here_data.duration is None:
+            return None
+
+        res = {}
+        res[ATTR_ATTRIBUTION] = self._here_data.attribution
+        res[ATTR_DURATION] = self._here_data.duration
+        res[ATTR_DISTANCE] = self._here_data.distance
+        res[CONF_UNIT_SYSTEM] = self._here_data.units
+        res['duration_without_traffic'] = self._here_data.base_time
+        res['origin_name'] = self._here_data.origin_name
+        res['destination_name'] = self._here_data.destination_name
+        res[CONF_MODE] = self._here_data.travel_mode
+        res[CONF_TRAFFIC_MODE] = self._here_data.traffic_mode
+        return res
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend depending on travel_mode."""
+        if self._here_data.travel_mode == TRAVEL_MODE_PEDESTRIAN:
+            return ICON_PEDESTRIAN
+        if self._here_data.travel_mode == TRAVEL_MODE_PUBLIC:
+            return ICON_PUBLIC
+        if self._here_data.travel_mode == TRAVEL_MODE_TRUCK:
+            return ICON_TRUCK
+        return ICON_CAR
+
+    def update(self):
+        """Update Sensor Information."""
+        # Convert device_trackers to HERE friendly location
+        if hasattr(self, '_origin_entity_id'):
```

Comment:
Check for `None` instead of using `hasattr`.

Reply:
done